HIP-1.md

```diff
@@ -0,0 +1,105 @@
+
+Furthermore, the modularity of sovereign consensus allows Hyperlane to be forward compatible with new security models as they emerge.
+
+### **Tech Spec**
+
+Sovereign consensus is powered by `Interchain Security Modules` or `ISMs`, which specify the rules that should be used when verifying an interchain message.
+
+`ISMs` must implement the folling interface:
+
+```solidity
+interface IInterchainSecurityModule {
+    /**
+     * @notice Returns an enum that represents the type of security model
+     * encoded by this ISM.
+     * @dev Relayers infer how to fetch and format metadata.
+     */
+    function moduleType() external view returns (uint8);
+
+    /**
+     * @notice Defines a security model responsible for verifying interchain
+     * messages based on the provided metadata.
+     * @param _metadata Off-chain metadata provided by a relayer, specific to
+     * the security model encoded by the module (e.g. validator signatures)
+     * @param _message Hyperlane encoded interchain message
+     * @return True if the message was verified
+     */
+    function verify(
+        bytes calldata _metadata,
+        bytes calldata _message
+    ) external returns (bool);
+}
+```
+
+Message `recipients` should specify their `ISM` via the `interchainSecurityModule()` view function.
+
+```solidity
+interface ISpecifiesInterchainSecurityModule {
+    /**
+     * @notice Returns the ISM to use for this recipient of interchain messages.
+     * @dev Optional, if not implemented the Mailbox's default ISM will be used.
+     */
+    function interchainSecurityModule() external view returns (IInterchainSecurityModule);
+}
+```
+
+The `Mailbox` contract, when delivering a message via `Mailbox.process()`, must check to see if the message recipient implements the `interchainSecurityModule()` view function. If it does, and returns a non-zero address, the `Mailbox` must call `verify()` on that address. Otherwise, it must call `verify()` on the `Mailbox's` default `ISM`.
+
+### **Rationale**
+
+Sovereign consensus allows for a modular approach to interchain security.
+This allows the core protocol to be forward compatible, ensuring that Hyperlane can support new security models as they are developed.
+Furthermore, it allows applications to select, configure, or invent security models that offer the most appropriate tradeoffs.
+
+We discuss a few of the design decisions made below.
+
+#### Finite ISM types
+
+This proposal relys on their being a finite set of ISM types that the relayer knows how to fetch and format metadata for.
+
+Alternatively, one could imagine a CCIP-read (or similar) based protocol for expressing arbitrary metadata, allowing for the definition of arbitrary ISMs.
```

Comment:
IMO, it is still a desirable to medium-term support arbitrary ISMs and their easy roll out to existing relayers, but like you mentioned, finite ISM types do not really prevent that from happening

Reply:
Let's take a "wait and see" approach here